### PR TITLE
Adjust tests to pass for both client-side and server-side validation

### PR DESCRIPTION
**Pull request by kevindelgado**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Expands validation tests to pass for when either client-side and server-side validation is enabled by default, in order to pass kubectl skew tests

See the [comment](https://github.com/kubernetes/kubernetes/pull/108350#issuecomment-1078236830) from #108350 and the related PR #108889 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by kevindelgado** [comment]
Everything is green across https://github.com/kubernetes/kubernetes/pull/108350 https://github.com/kubernetes/kubernetes/pull/108889 https://github.com/kubernetes/kubernetes/pull/109041 
PTAL @liggitt

**Comment by kevindelgado** [comment]
All feedback addressed and PRs synced

File: test/cmd/create.sh [hunk 1]
```diff
@@ -72,18 +72,6 @@ run_kubectl_create_error_tests() {
   # Passing no arguments to create is an error
   ! kubectl create || exit 1
 
-  ## kubectl create should not panic on empty string lists in a template
-  ERROR_FILE="${KUBE_TEMP}/validation-error"
-  kubectl create -f hack/testdata/invalid-rc-with-empty-args.yaml "${kube_flags[@]}" 2> "${ERROR_FILE}" || true
-  # Post-condition: should get an error reporting the empty string
-  if grep -q "unknown object type \"nil\" in ReplicationController" "${ERROR_FILE}"; then
```
Comment (liggitt) [inline test/cmd/create.sh:79]: this was a false positive where client-side validation failed something server-side validation would understand and persist. removing it is correct.

File: test/e2e/kubectl/kubectl.go [hunk 2]
```diff
@@ -1072,7 +1072,9 @@ metadata:
 			framework.ExpectNotEqual(schema, nil, "retrieving a schema for the crd")
 
 			meta := fmt.Sprintf(metaPattern, crd.Crd.Spec.Names.Kind, crd.Crd.Spec.Group, crd.Crd.Spec.Versions[0].Name, "test-cr")
-			validArbitraryCR := fmt.Sprintf(`{%s,"spec":{"bars":[{"name":"test-bar"}],"extraProperty":"arbitrary-value"}}`, meta)
```
Comment (liggitt) [inline test/e2e/kubectl/kubectl.go:1075]: this was a false-negative where client-side validation didn't know how to validate any properties in a custom resource that included x-kubernetes-preserve-unknown-fields anywhere. the adjusted fixture is an unknown field that both client-side and server-side validation will complain about

**Comment by liggitt** [comment]
this is needed for upgrade tests (1.23 tests running on upgraded 1.24 cluster + 1.24 kubectl)

**Comment by liggitt** [comment]
https://testgrid.k8s.io/google-gce-upgrade#gce-stable1-latest-upgrade-master-parallel&width=20&include-filter-by-regex=validation

**Comment by liggitt** [comment]
Screen Shot 2022-03-30 at 11 20 30 PM

**Pull request merged**